src/qz/printer/status/WmiPrinterStatusThread.java

```diff
@@ -16,18 +22,52 @@
     private boolean closing = false;
     private final String printerName;
     private final Winspool spool = Winspool.INSTANCE;
-    private int lastStatus = -1;
+    private int lastPrinterStatus = -1;
 
     private WinNT.HANDLE hChangeObject;
     private WinDef.DWORDByReference pdwChangeResult;
 
+    private HashMap<Integer, String> docNames = new HashMap<>();
+    private HashMap<Integer, ArrayList<Integer>> pendingJobStatuses = new HashMap<>();
+    private HashMap<Integer, Integer> lastJobStatusCodes = new HashMap<>();
+
+    Winspool.PRINTER_NOTIFY_OPTIONS listenOptions;
+    Winspool.PRINTER_NOTIFY_OPTIONS statusOptions;
+
     public WmiPrinterStatusThread(String name) {
         super("Printer Status Monitor " + name);
         printerName = name;
+
+        listenOptions = new Winspool.PRINTER_NOTIFY_OPTIONS();
+        listenOptions.Version = 2;
+        listenOptions.Flags = Winspool.PRINTER_NOTIFY_OPTIONS_REFRESH;
+        listenOptions.Count = 2;
+
+        Winspool.PRINTER_NOTIFY_OPTIONS_TYPE.ByReference[] mem = (Winspool.PRINTER_NOTIFY_OPTIONS_TYPE.ByReference[])
+                new Winspool.PRINTER_NOTIFY_OPTIONS_TYPE.ByReference().toArray(2);
+        mem[0].Type = Winspool.JOB_NOTIFY_TYPE;
+        mem[0].setFields(new short[] { JOB_NOTIFY_FIELD_STATUS, JOB_NOTIFY_FIELD_DOCUMENT });
+        mem[1].Type = Winspool.PRINTER_NOTIFY_TYPE;
+        mem[1].setFields(new short[] { PRINTER_NOTIFY_FIELD_STATUS });
+        listenOptions.pTypes = mem[0];
+
+        statusOptions = new Winspool.PRINTER_NOTIFY_OPTIONS();
+        statusOptions.Version = 2;
+        //statusOptions.Flags = Winspool.PRINTER_NOTIFY_OPTIONS_REFRESH;
```

Comment:
@Vzor- if you'd like to keep this around in that code, that's fine, just please leave an explanation as to why it's disabled.